### PR TITLE
portal: Make Keyring public and add helpful methods

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Also rename the helpers to match the libsecret names to ease the migration

### CODE REVIEW

File: src/portal/mod.rs [hunk 1]
```diff
@@ -41,8 +43,10 @@ mod secret;
 
 pub use error::Error;
 
-struct Keyring {
-    keyring: api::Keyring,
+pub use self::api::Item;
+
+pub struct Keyring {
```
[the remark this answers is not in the document]
Comment (sophie-h) [reply src/portal/mod.rs:48]: > without having to deal with the keys manually

Do you mean getting the key from the portal? If you are doing anything password manager like you have to get all the separate keys from the host's secret-service manually and use them for each app respectively.

File: src/keyring.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            items
+                .into_iter()
+                .map(|i| Item::for_file(i, self.is_sandboxed))
+                .collect::<Vec<_>>()
+        };
+        Ok(items)
+    }
+}
+
+pub struct Item {
```
Comment (sophie-h) [inline src/keyring.rs:116]: Any reason not to use an enum here?

File: src/keyring.rs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+use crate::{
+    dbus::{self, Algorithm, DEFAULT_COLLECTION},
+    portal, Result,
+};
+
+/// A generic implementation of Keyring.
+///
+/// It would automatically use the file backend if the application is sandboxed
+/// and otherwise falls back to the DBus service.
+pub struct Keyring {
```
Comment (sophie-h) [inline src/keyring.rs:15]: Why not enum?

Comment (bilelmoussaoui) [reply src/keyring.rs:15]: Because it contains the is_sandboxed field but sure, an enum could work indeed

File: src/keyring.rs (hunk 2 above)
Comment (bilelmoussaoui) [reply src/keyring.rs:116]: will switch to that

File: src/portal/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/portal/mod.rs:48]: > Do you mean getting the key from the portal?

Yes, the lower level api allows you to deal with the keys yourself for a use case like implementing seahorse alternative. But if I don't want to ever fallback to dbus for whatever reason, having Keyring public is useful as I don't have to get the key manually from the portal or having to remember I should call .dump or decrypt my items.

[the remark this answers is not in the document]
Comment (sophie-h) [reply src/portal/mod.rs:48]: I think the helpers should cover most of the cases but if you see a use case for it, I'm fine with having it public

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/portal/mod.rs:48]: I don't really have a use case actually... but let is figure out that later

**Pull request merged**